hnn_core/tests/test_gui.py

```diff
@@ -85,49 +91,67 @@ def _import_json(content):
+
+    # check parameters with different files
+    assert gui.connectivity_widgets[0][0].children[1].value == 0.02
+    # value should change when loading connectivity from file 2
+    gui._simulate_upload_connectivity(file2_path)
+    assert gui.connectivity_widgets[0][0].children[1].value == 0.01
+
+    # Load drives and make sure connectivity does not change
+    gui._simulate_upload_drives(file1_path)
+    assert gui.connectivity_widgets[0][0].children[1].value == 0.01
+
+    plt.close('all')
+
+
+def test_gui_upload_drives():
+    """Test if gui handles uploaded drive parameters correctly"""
     gui = HNNGUI()
     _ = gui.compose()
 
-    # change the default loaded parameters
+    # clear the drive widgets
     original_drive_count = len(gui.drive_widgets)
     assert original_drive_count > 0
     gui.delete_drive_button.click()
     assert len(gui.drive_widgets) == 0
 
-    original_tstop = gui.widget_tstop.value
-    gui.widget_tstop.value = 1
-    original_tstep = gui.widget_dt.value
-    gui.widget_dt.value = 1
     # simulate upload default.json
-    file1_url = "https://raw.githubusercontent.com/jonescompneurolab/hnn-core/master/hnn_core/param/default.json"  # noqa
-    file2_url = "https://raw.githubusercontent.com/jonescompneurolab/hnn-core/master/hnn_core/param/gamma_L5weak_L2weak.json"  # noqa
-    gui._simulate_upload_connectivity(file1_url)
-    gui._simulate_upload_drives(file1_url)
+    file1_url = Path(hnn_core_root, 'param', 'jones2009_base.json')
+    file2_url = Path(assets_path, 'gamma_L5weak_L2weak_hierarchical.json')
 
-    # check if parameter is reloaded.
-    assert gui.widget_tstop.value == original_tstop
-    assert gui.widget_dt.value == original_tstep
+    # check if parameter reloads
+    gui._simulate_upload_drives(file1_url)
     assert len(gui.drive_widgets) == original_drive_count
+    drive_types = [widget['type'] for widget in gui.drive_widgets]
+    assert drive_types == ['Evoked', 'Evoked', 'Evoked']
 
     # check parameters with different files.
-    # file1: connectivity file2: drives
-    gui._simulate_upload_connectivity(file1_url)
-    assert gui.widget_tstop.value == 170.
-    assert gui.connectivity_widgets[0][0].children[1].value == 0.02
     gui._simulate_upload_drives(file2_url)
-    assert gui.widget_tstop.value == 250.
-    # uploading new drives does not influence the existing connectivity.
-    assert gui.connectivity_widgets[0][0].children[1].value == 0.02
+    assert len(gui.drive_widgets) == 1
+    assert gui.drive_widgets[0]['type'] == 'Poisson'
+
+    # Load connectivity and make sure drives did not change
+    gui._simulate_upload_connectivity(file1_url)
+    assert len(gui.drive_widgets) == 1
 
-    # file2: connectivity file1: drives
-    gui._simulate_upload_connectivity(file2_url)
-    # now connectivity is refreshed.
-    assert gui.connectivity_widgets[0][0].children[1].value == 0.01
-    assert gui.drive_widgets[-1]['tstop'].value == 250.
```

Comment:
this would be a good one to keep for the poisson drive in `file2` to make sure it's loaded properly

Reply:
The weight check is currently preserved on line 116. However the stop time check is not because tstop isn't currently saved with the Network configurations.

Reply:
I was thinking that this was referring to the stop time of the drive?

Reply:
oh you're right! hmm... I can add it back in. I'm curious if the drive widget will set it to the tstop widget if the config tstop is greater than the the widget...

Reply:
I added the test but set it to 170 instead of 250. The logic sets the drive tstop to the GUI's widget tstop value if the drive tstop value is larger. We currently don't save a universal tstop value to Network object, so the GUI doesn't set its tstop widget from Network configuration files. 

This may change in the future if we want to save a stop value with a Network.